### ISSUE TITLE
Add support for Jaegar schema [FEATURE]

### ISSUE THREAD

**Issue by anirudha** [opensearch-project/observability#755]
**Is your feature request related to a problem?**
A clear and concise description of what the problem is, e.g. _I'm always frustrated when [...]_

**What solution would you like?**

**What alternatives have you considered?**

**Do you have any additional context?**

**Comment by afishler** [comment]
Jaeger is a very popular Open Source project for distributed tracing. It is often that a Jaeger ingestion architecture is created as an infrastructure and the whole span ingestion pipeline is already written in Jaeger format which is different from OpenTelemetry format.

It would provide a lot of value for users who are already using OpenSearch as their tracing backend but with Jaeger to be able to quickly add or migrate to the OSD trace analytics by just adding the observability plugin and not investing in several ingestion methods or formats.

For that, OSD observability plugin can support out of the box Jaeger index format and based on configuration can be used to read also Jaeger spans.

Types of data that require handling is:

- Span data
- Service dependency data
- Service and Operation (endpoints) data

The Jaeger struct is defined here
https://github.com/jaegertracing/jaeger/blob/v1.33.0/plugin/storage/es/spanstore/dbmodel/model.go#L48-L67

Logz.io uses a slightly modified structure in some aspects
https://github.com/logzio/jaeger-logzio/blob/master/store/objects/logzioSpan.go#L16-L32

**Comment by jkowall** [comment]
I have had this come up with several users of OSS and AWS managed OpenSearch. Is there a timeline on supporting Jaeger schema to avoid having to store the data twice? Thanks!

**Comment by anirudha** [comment]
yes, we will post a sample demo for feedback and plan a release on this. @derek-ho @jkowall

**Comment by derek-ho** [comment]
@jkowall @anirudha @kavck here is a quick video demo of trace analytics working with raw jaeger data... you can watch it sped up to save some time.

https://user-images.githubusercontent.com/26328171/204663801-bc068a33-4199-4186-93a7-ed1af139aeb4.mov

**Comment by derek-ho** [comment]
A few discussion points:
- We cannot generate the service map without running a very expensive query - should we take this out of the Jaeger version of trace analytics?
- There is no such thing as trace group in jaeger world @jkowall do you see a valid alternative that we can show on the dashboard page (for example operationName, or operationName combined with serviceName)? If not this landing page maybe should be removed as well, since it is not providing any value....
- Any other feature gap between this and jaeger that you see?
- UX experience between navigating this raw jaeger experience and data prepper experience - @kavck (Discover dropdown to choose between the two modes)?

**Comment by jkowall** [comment]
This is really nice! I wanted to point out that the way the topology is generated is via Spark jobs typically which summarize the data. My last company also ported that to Kafka streams. The product manager at AWS for this feature is @RajivTaori who should have a look at it. I'm also notifying @elfisher who is another PM leader for the project. I think this is something that would be useful to the community.

We could potentially solve the Jaeger issue for topology in several other ways, but I'm not sure it would be a great idea to rebuild around that. We could probably process the Jaeger data in the format that OpenSearch trace analytics expects.

**Comment by jkowall** [comment]
In Jaeger this is how it works (Spark), ideally we wanted to move it to use the service map processor on Otel, but never happened.

**Comment by elfisher** [comment]
hey @derek-ho can we get a PR opened up and start working off of it? Getting Jaeger support is really exciting.

**Comment by derek-ho** [comment]
@elfisher I opened a draft PR with the work done so far. Looping @kavck in for UX support and will update here when more progress is done.

**Comment by kavck** [comment]
@derek-ho here's an initial draft to continue the discussion (demo and screens included below):

### Traces Overview - Workflow Draft 1

![tracesOverviewDraft1](https://user-images.githubusercontent.com/108144351/208472867-63df01d9-3af3-442d-8fb4-14a2d34ff5ee.gif)

<details>
<summary>Screen by screen mockups</summary>
<p>

![tracesDraft1-a](https://user-images.githubusercontent.com/108144351/208472926-6903a61d-19df-479b-b051-fabdf71f86f8.png)

![tracesDraft1-b](https://user-images.githubusercontent.com/108144351/208472943-e187f356-d957-4cdf-935b-581dd54cceac.png)

![tracesDraft1-c](https://user-images.githubusercontent.com/108144351/208472961-9a4a0d9d-d507-4e0b-ac7a-863939015af0.png)

</p>
</details>

**Comment by derek-ho** [comment]
@elfisher @jkowall any thoughts on this mock up? Have already started development on it, but I can incorporate any feedback you folks have

**Comment by kgcreative** [comment]
Help me understand why we have separate graphs for error rate, latency and throughput, when the spans in each also have error rates, and latency? Wouldn't it make more sense to add a throughput column to the table, and allow users to filter by each of those ascending/descending? That would give us the same "top N" view, but in a way that wouldn't require the user to expand or collapse three sepparate graphs.

**Comment by derek-ho** [comment]
https://user-images.githubusercontent.com/26328171/209233189-dd8df408-f6e3-4d41-8728-73b8a04d5625.mov

**Comment by derek-ho** [comment]
> Screen.Recording.2022-12-20.at.4.42.52.PM.mov

Still ironing out a few connection points and figuring out which groups make sense to put on the dashboard page...

**Comment by kavck** [comment]
@kgcreative that's a great idea! That approach could bring all the graphs together in a single table. What we understand, however, about the individual graphs is that they are useful to quickly spot interesting signals for the user to deep dive on anything of interest to them.

@derek-ho a continuation of some iterations is shown in the draft below:
![tracesDraft2](https://user-images.githubusercontent.com/108144351/209368862-a1bbf489-72a4-4fa1-bcba-ff1185ed98c7.png)

**Comment by kgcreative** [comment]
@kavck there is also an opportunity to perhaps display these either side by side, or using a segmented control to switch modes. I'm not convinced that a long vertical scroll necessarily will make sense to our customers, particularly since the table seems to be what's more useful to help augment the viz info. It would be worth exploring in that direction as well

**Comment by derek-ho** [comment]
Updated video of work: 

https://user-images.githubusercontent.com/26328171/209825900-ac65316b-f4b6-4f72-ab8b-5c03c1a2fdd4.mov


I believe it is mostly feature complete for now for the next release, but we can make UI/UX improvements on top of it. I will repost this to the Jaeger community so we can get their input and make a roadmap ahead for future releases. 

Updated Figma: https://www.figma.com/file/TfdCoPpEEhdwuFXf5R2UC5/%F0%9F%9F%A2-Traces-Workspace?node-id=3759%3A51379&t=Rthmgo8EeQokn6mU-1

Caveat: This demo is made from jaeger data ingested in a specific format that is more friendly to opensearch: 
```--es.tags-as-fields.all=true```. It doesn't work well if it is not ingested in this format, we may want to bring this up to the community or make this the default, since it is not today. 

@anirudha @elfisher @kavck @jkowall

**Comment by derek-ho** [comment]
@kavck screenshots from the toggle view

![Screen Shot 2022-12-28 at 2 34 12 PM](https://user-images.githubusercontent.com/26328171/209863397-b5d09d60-0a2d-4eb8-9be3-d175728f41bc.png)
![Screen Shot 2022-12-28 at 2 34 05 PM](https://user-images.githubusercontent.com/26328171/209863398-00590a36-678e-4152-8190-6a2a506d4893.png)

**Comment by derek-ho** [comment]
Updated images from what the feature looks like today - based on UI/UX suggestions

<img width="1351" alt="Screen Shot 2023-01-03 at 9 28 05 AM" src="https://user-images.githubusercontent.com/26328171/210376975-ab8b0cfc-9e0d-4fe9-b430-641a3ee57bd4.png">
<img width="1313" alt="Screen Shot 2023-01-03 at 9 27 50 AM" src="https://user-images.githubusercontent.com/26328171/210376983-de7beaa3-4784-4364-8c89-94e2807c6929.png">
